trade-finance-manager-ui/server/controllers/facilities/index.test.js

```diff
@@ -1,137 +1,440 @@
-import caseController from '.';
+import facilityController from '.';
 import api from '../../api';
-import { mockRes } from '../../test-mocks';
-import CONSTANTS from '../../constants';
+import PageOutOfBoundsError from '../../errors/page-out-of-bounds.error';
+import { mockRes as generateMockRes } from '../../test-mocks';
+
+const CONSTANTS = require('../../constants');
 
 describe('controllers - facilities', () => {
-  let res;
+  let mockRes;
+  const mockReqTemplate = {
+    session: {
+      user: {},
+      userToken: 'userToken',
+    },
+    body: {},
+    params: {},
+    query: {},
+  };
   const mockFacilities = [
     {
-      dealId: '121212',
-      facilityId: '112233',
-      ukefFacilityId: '123456',
-      type: CONSTANTS.FACILITY.FACILITY_TYPE.CASH,
-      companyName: 'Mock Company Name',
-      value: 'GBP 1,000,000.00',
-      coverEndDate: '2021-08-12T00:00:00.000Z',
-      dealType: CONSTANTS.DEAL.DEAL_TYPE.GEF,
-      hasBeenIssued: true,
-      currency: 'GBP',
+      facilityId: '0',
+      hasAmendmentInProgress: false,
     },
     {
-      dealId: '122122',
-      facilityId: '223344',
-      ukefFacilityId: '1234567',
-      type: CONSTANTS.FACILITY.FACILITY_TYPE.CASH,
-      companyName: 'Mock Company Name',
-      value: 'EUR 18,000,000.00',
-      coverEndDate: null,
-      dealType: CONSTANTS.DEAL.DEAL_TYPE.BSS_EWCS,
-      hasBeenIssued: false,
-      currency: 'EUR',
+      facilityId: '1',
+      hasAmendmentInProgress: false,
     },
   ];
-
-  const mockGetFacilities = { facilities: mockFacilities };
+  const mockApiGetFacilitiesResponse = {
+    facilities: mockFacilities,
+    pagination: {
+      totalItems: mockFacilities.length,
+      currentPage: 0,
+      totalPages: 1,
+    },
+  };
+  const mockAmendments = [
+    {
+      status: CONSTANTS.AMENDMENTS.AMENDMENT_STATUS.IN_PROGRESS,
+      facilityId: '0',
+    },
+    {
+      status: CONSTANTS.AMENDMENTS.AMENDMENT_STATUS.NOT_STARTED,
+      facilityId: '1',
+    },
+  ];
+  const mockApiGetAllAmendmentsInProgressResponse = {
+    data: mockAmendments,
+  };
 
   beforeEach(() => {
-    res = mockRes();
+    mockRes = generateMockRes();
   });
 
-  describe('GET all facilities', () => {
+  describe('GET facilities', () => {
     describe('when there are facilities', () => {
       beforeEach(() => {
-        api.getFacilities = () => Promise.resolve(mockGetFacilities);
-        api.getAllAmendmentsInProgress = () => Promise.resolve({ data: [] });
+        api.getFacilities = jest.fn().mockImplementation(() => Promise.resolve(mockApiGetFacilitiesResponse));
+        api.getAllAmendmentsInProgress = jest.fn().mockImplementation(() => Promise.resolve({}));
+      });
+
+      describe('when the pageNumber, sortfield, sortorder and search are not specified in the request', () => {
+        const mockReq = JSON.parse(JSON.stringify(mockReqTemplate));
```

Comment:
Ooh, I didn't know about JSON.Parse being an antipattern - good to know, thank you!

I assume you're referring to `lodash` `cloneDeep` - is there any reason not to use the native [structuredClone](https://developer.mozilla.org/en-US/docs/Web/API/structuredClone) which is supported in Node 17 and later, rather than introducing another dependency?

I've used `structuredClone` for now, but I'm happy to change this